src/features/courses/course-controller.ts

```diff
@@ -3271,6 +3272,63 @@ class CourseController {
             return { problemScores: problemScoresReturn, bestVersionScore: bestVersionScoreReturn, bestOverallVersion: bestOverallVersionReturn};
         });
     };
+    async canUserGradeAssessment({
+        user,
+        topicId
+    }: CanUserGradeAssessmentOptions): Promise<boolean> {
+        const topic = await courseRepository.getCourseTopic({ id: topicId });
+        const unit = await courseRepository.getCourseUnit({ id: topic.courseUnitContentId });
+        const course = await courseRepository.getCourse({ id: unit.courseId });
+        return (user.id === course.instructorId);
+    }
+
+    async getAssessmentForGrading({
+        topicId
+    }: GetAssessmentForGradingOptions): Promise<GetAssessmentForGradingResult> {
+        // eventually update to include different parameters
+        // grade best individual problems // grade best version attempt // get ALL versions?!?
+        try {
+            const topic = await CourseTopicContent.findOne({
+                where: {
+                    id: topicId,
+                    active: true,
+                },
+            });
+
+            if (_.isNil(topic)) {
+                throw new WrappedError(`failed to retrieve topic #${topicId} for grading`);
+            }
+
+            const problems = await CourseWWTopicQuestion.findAll({
+                where: {
+                    courseTopicContentId: topicId,
+                    active: true
```

Comment:
Do we need to account for hidden? We have the flag in the db but no way to use it now